File: app/src/main/java/com/master/kit/TestCase/PullToRefreshActivity.java
package com.master.kit.testcase;

import android.content.Context;
import android.os.Bundle;
import android.os.Handler;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.BaseAdapter;
import android.widget.TextView;

import com.master.kit.R;
import com.ooftf.pulltorefresh.widget.PullToLoadMoreFooter;
import com.ooftf.pulltorefresh.widget.PullToRefreshHeader;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

import static android.R.id.content;
import static android.R.id.list;

public class PullToRefreshActivity extends AppCompatActivity {

    @BindView(R.id.main_listview)
    com.ooftf.pulltorefresh.widget.PullToRefreshListView mainListview;
    Handler handler;

    MyAdapter myAdapter;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_pull_to_refresh);
        ButterKnife.bind(this);
        myAdapter = new MyAdapter(this);
        mainListview.setAdapter(myAdapter);
        for(int i =0;i<100;i++){
            myAdapter.getDatas().add(i+"");
        }
        handler = new Handler();
        final PullToLoadMoreFooter pullToLoadMoreFooter = new PullToLoadMoreFooter(this);
        pullToLoadMoreFooter.setListView(mainListview);
        pullToLoadMoreFooter.setOnLoadingMoreListener(new PullToLoadMoreFooter.OnLoadingMoreListener() {
            @Override
            public void OnLoadingMore() {
                handler.postDelayed(new Runnable() {
                    @Override
                    public void run() {
                        for(int i =0;i<100;i++){
                            myAdapter.getDatas().add(i+"");
                        }
                        pullToLoadMoreFooter.loadingComplete();
                    }
                },5000);
            }
        });
        mainListview.setOnRefreshListener(new PullToRefreshHeader.OnRefreshListener() {
            @Override
            public void onRefreshing() {
                handler.postDelayed(new Runnable() {
                    @Override
                    public void run() {
                        mainListview.onRefreshComplete();
                        pullToLoadMoreFooter.hasMore();
                    }
                },5000);
            }
        });
        mainListview.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                Log.e("OnItemClickListener","position::"+position);
            }
        });

    }
    class MyAdapter extends BaseAdapter{
        List<String> datas;
        Context context;
        MyAdapter(Context context){
            this.context = context;
            datas = new ArrayList<>();
        }
        @Override
        public int getCount() {
            return datas.size();
        }

        @Override
        public Object getItem(int position) {
            return datas.get(position);
        }

        @Override
        public long getItemId(int position) {
            return position;
        }

        @Override
        public View getView(int position, View convertView, ViewGroup parent) {
            TextView view = (TextView) convertView;
            if(view == null){
                view = new TextView(context);
            }
            view.setText(datas.get(position));
            return view;
        }

        public List<String> getDatas() {
            return datas;
        }
    }

}

File: self-made/src/main/java/com/ooftf/widget/activity/PullToRefreshActivity.kt
package com.ooftf.widget.activity

import android.content.Context
import android.os.Bundle
import android.os.Handler
import android.util.Log
import android.view.View
import android.view.ViewGroup
import android.widget.AdapterView
import android.widget.BaseAdapter
import android.widget.TextView
import com.alibaba.android.arouter.facade.annotation.Route
import com.ooftf.service.base.BaseSlidingActivity
import com.ooftf.widget.R
import com.ooftf.widget.self.pulltoloading.PullToLoadingLayout
import com.ooftf.widget.self.pulltoloading.PullToLoadingView
import kotlinx.android.synthetic.main.activity_pull_to_refresh.*
import java.util.*

@Route(path = "/widget/pullToRefresh")
class PullToRefreshActivity : BaseSlidingActivity() {

    private var handler = Handler()
    val adapter by lazy {
        MyAdapter(this)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_pull_to_refresh)
        listView.adapter = adapter
        fillData()
        val pullToLoadMoreFooter = PullToLoadingLayout(this, PullToLoadingView(this), false)
        pullToLoadMoreFooter.setListView(listView)
        pullToLoadMoreFooter.setLoadEvent { simulateLoadingMore(pullToLoadMoreFooter) }
        pullToRefreshRoot.setOnRefreshListener {
            simulatePullDownRefresh(pullToLoadMoreFooter)
        }
        listView.onItemClickListener = AdapterView.OnItemClickListener { _, _, position, _ -> Log.e("OnItemClickListener", "position::" + position) }

    }

    private fun simulatePullDownRefresh(pullToLoadMoreFooter: PullToLoadingLayout) {
        handler.postDelayed({
            pullToRefreshRoot.onRefreshComplete()
            pullToLoadMoreFooter.normal()
        }, 5000)
    }

    private fun simulateLoadingMore(pullToLoadMoreFooter: PullToLoadingLayout) {
        handler.postDelayed({
            fillData()
            if (switchView.isChecked) {
                pullToLoadMoreFooter.error()
            } else {
                pullToLoadMoreFooter.normal()
            }
        }, 5000)
    }

    private fun fillData() {
        for (i in 0..99) {
            adapter.data.add(i.toString() + "")
        }
        adapter.notifyDataSetChanged()
    }

    class MyAdapter(var context: Context) : BaseAdapter() {
        var data: MutableList<String> = ArrayList()

        override fun getCount(): Int {
            return data.size
        }

        override fun getItem(position: Int): Any {
            return data[position]
        }

        override fun getItemId(position: Int): Long {
            return position.toLong()
        }

        override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
            var view: TextView = if (convertView == null) {
                TextView(context)
            } else {
                convertView as TextView
            }
            view.text = data[position]
            return view
        }
    }

}
